index.rst

```diff
@@ -46,14 +46,314 @@
+In addition, there is also the question of how those additional resources are handled by the application and the programming language itself.
+For instance, Python does not easily support multiprocessing, so DDS services will be competing for the same computing resources as those needed by the application to run properly.
+
+.. figure:: /_static/SingleProcessFig.png
+   :name: fig-single-process
+   :target: ../_images/SingleProcessFig.png
+   :alt: DDS single process mode
+
+   DDS architecture when running in single process mode.
+   In the left hand panel, a diagram showing the internal architecture of a single application.
+   The "DDS Application Process" represents the software developed on top of SAL/SalObj to communicate over DDS and execute some operation.
+   In the case of Rubin Observatory system this would be a CSC, salkafka producer, user notebook server, etc.
+   In the right hand panel, a high level overview of a group of applications (CSCs, salkafka producers, etc).
+   Each application runs their own DDS services alongside their processes.
+
+As one can imagine, this architecture will not scale well for large scale systems.
+If the DDS communication grows large enough (in terms of number of topics and published frequency) the processing requirements for the DDS services alone may start to interfere with the application and vice-versa.
+To overcome this problem, OpenSplice DDS can be configured to run in shared memory mode.
+
+When running in shared memory mode, the user configures and runs a single daemon per node that is dedicated to running the OpenSplice services.
+Applications running in the same node can then be configured to attach to the daemon using system shared memory.
+In this case, several applications share the same DDS services.
+The computing resources needed by the DDS services are separated from those of the application and will not interfere with each other (assuming the node has enough resources available).
+
+.. figure:: /_static/SharedMemoryFig.png
+  :name: fig-shared-memory
+  :target: ../_images/SharedMemoryFig.png
+  :alt: DDS shared memory mode
+
+  DDS architecture when running in shared memory mode.
+  In the left hand panel, a diagram showing the internal architecture of several applications (denoted "DDS Application Processes") running concurrently in the same node.
+  All applications share the same OpenSplice DDS services through the shared memory space.
+  In the right hand panel, a high level overview of a group of applications running in the same node (CSCs, salkafka producers, etc) sharing the OpenSplice DDS services.
+
+
+.. _NCSA-Standard-Deployment-Test:
+
+NCSA Standard Deployment Test
+=============================
+
+This test consisted of trying to reproduce the failure mode observed in multiple occasions at the NCSA test stand when running the MT components.
```

Comment:
the failure mode here is *only* the long startup times?  Or other aspects as well... meaning the notebooks never start etc.

I think it'd be good to add a sentence somewhere (probably towards the top) saying what the user experiences when failures happen.

Reply:
They are all the same problem.. the notebook "never start" problem was because the timeout was set to 300s instead of the normal 60s... It would eventually timeout as well, if given enough time.. 

In the overview section I mention what this new failure is.. but I'll improve it a bit there and mention it here.